src/components/BookForm.tsx

```diff
@@ -0,0 +1,104 @@
+"use client";
+import AutoForm from "@/components/ui/auto-form";
+import * as z from "zod";
+
+import { Card, CardContent, CardHeader, CardTitle } from "./ui/card";
+import { Button } from "@/components/ui/button";
+import { api } from "~/utils/api";
+import React, { useState } from "react";
+import {
+  Popover,
+  PopoverContent,
+  PopoverTrigger,
+} from "@/components/ui/popover";
+
+const bookInputs = z.object({
```

Comment:
zod schema's are always postfixed with 'schema`, so in this case we would call it 
```suggestion
const BookInputSchema = z.object({
```